core/payments/payments.go

```diff
@@ -769,8 +773,10 @@ func (s *service) getAllTiers(ctx context.Context, req *pb.RpcMembershipGetTiers
 
 	// if NoCache -> skip returning from cache
 	if !req.NoCache && (err == nil) && (cachedTiers != nil) && (cachedTiers.Tiers != nil) {
-		log.Debug("returning tiers from cache", zap.Error(err), zap.Any("cachedTiers", cachedTiers))
-		return cachedTiers, nil
+		if (cachedTiers.Error == nil) || (cachedTiers.Error.Code == pb.RpcMembershipGetTiersResponseError_NULL) {
```

Comment:
It seems we can simplify the beginning of func a little bit, as cachedTiers is used only when NoCache is false. So lines 770-780 could be:

```
	// 1 - check in cache in case NoCache is False
	if !req.NoCache {
		_, cachedTiers, err := s.cache.CacheGet()
		if err == nil && cachedTiers != nil && cachedTiers.Tiers != nil &&
			(cachedTiers.Error == nil || cachedTiers.Error.Code == pb.RpcMembershipGetTiersResponseError_NULL) {
			log.Debug("returning tiers from cache", zap.Error(err), zap.Any("cachedTiers", cachedTiers))
			return cachedTiers, nil
		}
	}
```
It is safe that we have check **cache==nil** and checks on cache fields in one logical expression, because Go quaranties sequental evaluation